File: app/src/main/java/org/xdty/callerinfo/data/CallerDataSource.java
package org.xdty.callerinfo.data;

import org.xdty.callerinfo.model.SearchMode;
import org.xdty.callerinfo.model.db.Caller;

import java.util.Map;

import rx.Observable;

public interface CallerDataSource {

    Caller getCallerFromCache(String number);

    Observable<Caller> getCaller(String number);

    Observable<Caller> getCaller(String number, boolean forceOffline);

    Observable<Map<String, Caller>> loadCallerMap();

    void setOnDataUpdateListener(OnDataUpdateListener listener);

    Observable<Void> clearCache();

    void updateCaller(String number, int type, String typeText);

    boolean isIgnoreContact(String number);

    SearchMode getSearchMode(String number);

    interface OnDataUpdateListener {

        void onDataUpdate(Caller caller);
    }

}

File: app/src/main/java/org/xdty/callerinfo/data/CallerDataSource.kt
package org.xdty.callerinfo.data

import io.reactivex.Observable
import org.xdty.callerinfo.model.SearchMode
import org.xdty.callerinfo.model.db.Caller

interface CallerDataSource {
    fun getCallerFromCache(number: String): Caller
    fun getCaller(number: String): Observable<Caller>
    fun getCaller(number: String, forceOffline: Boolean): Observable<Caller>
    fun loadCallerMap(): Observable<Map<String, Caller>>
    fun setOnDataUpdateListener(listener: OnDataUpdateListener)
    fun clearCache(): Observable<Int>
    fun updateCaller(number: String, type: Int, typeText: String)
    fun isIgnoreContact(number: String): Boolean
    fun getSearchMode(number: String): SearchMode

    interface OnDataUpdateListener {
        fun onDataUpdate(caller: Caller)
    }
}